### ISSUE TITLE
Doesn't AccountMapper violate the Single Responsibility Principle?

### ISSUE THREAD

**Issue by jaylensoeur** [thombergs/buckpal#11]
SRP (Single Responsibility Principles) states that "a module should have one reason to change".

I see you have multiple methods to map to different Entities in the one class. Doesn't this violate the SRP? The class now has multiple reasons to change. I know SRP is just a guideline, it would be great if you explained why you decided on the trade off.

https://github.com/thombergs/buckpal/blob/master/adapters/buckpal-persistence/src/main/java/io/reflectoring/buckpal/adapter/persistence/AccountMapper.java

**Comment by thombergs** [comment]
Hi @jaylensoeur 

good question.

Yes, the `AccountMapper` has to change when either the `Account` entity or the `ActivityWindow` entity changes. 

But, I decided that I don't care much about that :).

That's because I don't mind changing code in the persistence adapter if the domain model changes. I would mind very much if I had to change code in the domain model if the persistence adapter changes, however!

So, I decided that the direction of the dependency is more important to me than having a very narrow responsibility in a mapper class.

Saying it with Uncle Bob's words: the persistence adapter is "just a detail". The important stuff happens in the use case implementations and the domain model. This is where the business rules are implemented. The database "just" stores and retrieves data (and maps it into the domain model). So I don't care if I have to change the database if the business rules change.

Having said that, I would not put all mapping methods into a single mapper, but instead, group them into mappers by entities if there are more than a handful or so. But that's just standard code hygiene :).
